Cobra-Closet/mix-match
# Deployment: Merge Dev Branch into Main Branch

## Description:
This pull request encompasses substantial enhancements and additions to the Cobra Closet application. The changes implemented in this deployment from the dev branch to the main branch significantly improve the application's functionality, testing, and overall user experience. The major updates include:

### Building Frontend and Backend Testing Suites: 
Both frontend and backend testing suites have been established, leveraging Jest, Supertest, and React Testing Library (beta). These suites rigorously test the application's functionality, ensuring reliability across various scenarios.

### Creation of a New SQL Database: 
A new SQL database has been created to efficiently manage and store wardrobe items, outfit suggestions, and user-related data.

### Integration of Cloudinary API: 
The Cloudinary API has been integrated to facilitate the storage and management of images, enhancing the user experience by allowing users to associate images with their wardrobe items.

### Session Persistence with Cookies: 
Persistent sessions using cookies have been implemented, enhancing user convenience by maintaining logged-in states between sessions.

### Custom Clothing Item Addition Feature: 
Users can now add custom clothing items to their wardrobe, providing greater flexibility and personalization in managing their clothing inventory.

### Automated Session Cleanup: An additional feature of automated session cleanup has been incorporated, ensuring efficient and automatic session termination after a specified period, enhancing security and system performance.


These updates align with the application's aim to streamline wardrobe management, outfit selection, and user interaction. Additionally, they set the groundwork for future improvements, such as advanced recommendation algorithms based on user preferences and weather conditions.

## Testing and Deployment Instructions:
- Ensure all dependencies are installed by running npm install in the project root directory.
- Start the Express backend server by navigating to the backend folder and using npm start.
- Initiate the React frontend application by accessing the frontend folder and using npm start.
- Test the application's functionality using the established testing suites for both frontend and backend.

This pull request has undergone thorough testing and code review to ensure its quality and adherence to coding standards. Your feedback and suggestions for further improvements are greatly appreciated.
